lib/compiler-llvm/src/compiler.rs

```diff
@@ -80,110 +77,95 @@ impl LLVMCompiler {
-                    let module =
-                        func_trampoline.dynamic_trampoline_to_module(sig, self.config(), &name)?;
-                    Ok(module.write_bitcode_to_memory().as_slice().to_vec())
-                },
-            );
-
-        let merged_bitcode = merged_bitcode
-            .chain(trampolines_bitcode)
-            .chain(dynamic_trampolines_bitcode)
-            .collect::<Result<Vec<_>, CompileError>>()?
-            .into_par_iter()
-            .reduce_with(|bc1, bc2| {
-                let ctx = Context::create();
-                let membuf = MemoryBuffer::create_from_memory_range(&bc1, "");
-                let m1 = Module::parse_bitcode_from_buffer(&membuf, &ctx).unwrap();
-                let membuf = MemoryBuffer::create_from_memory_range(&bc2, "");
-                let m2 = Module::parse_bitcode_from_buffer(&membuf, &ctx).unwrap();
-                m1.link_in_module(m2).unwrap();
-                m1.write_bitcode_to_memory().as_slice().to_vec()
-            });
-        let merged_module = if let Some(bc) = merged_bitcode {
-            let membuf = MemoryBuffer::create_from_memory_range(&bc, "");
-            Module::parse_bitcode_from_buffer(&membuf, &ctx).unwrap()
-        } else {
-            ctx.create_module("")
-        };
-
-        let i8_ty = ctx.i8_type();
-        let metadata_init = i8_ty.const_array(
-            wasmer_metadata
-                .iter()
-                .map(|v| i8_ty.const_int(*v as u64, false))
-                .collect::<Vec<_>>()
-                .as_slice(),
+        let dynamic_trampolines = compile_info.module.functions.iter().par_bridge().map_init(
+            || {
+                let target_machine = self.config().target_machine(target);
+                (
+                    FuncTrampoline::new(target_machine),
+                    &compile_info.module.signatures,
+                )
+            },
+            |(func_trampoline, signatures), (i, sig)| {
+                let sig = &signatures[*sig];
+                let symbol = Symbol::DynamicFunctionTrampoline(i);
+                let name = symbol_registry.symbol_to_name(symbol);
+                let module =
+                    func_trampoline.dynamic_trampoline_to_module(sig, self.config(), &name)?;
+                let memory_buffer = func_trampoline
+                    .target_machine
+                    .write_to_memory_buffer(&module, FileType::Object)
+                    .unwrap();
+                Ok(memory_buffer.as_slice().to_vec())
+            },
         );
-        let metadata_gv =
-            merged_module.add_global(metadata_init.get_type(), None, "WASMER_METADATA");
-        metadata_gv.set_initializer(&metadata_init);
-        metadata_gv.set_linkage(Linkage::DLLExport);
-        metadata_gv.set_dll_storage_class(DLLStorageClass::Export);
-
-        if self.config().enable_verifier {
-            merged_module.verify().unwrap();
-        }
```

Comment:
This was not needed any longer since we already do verification on `translate_to_module`

Reply:
No action required. It was there to check the module after changing the IR with `merged_module.add_global`.